### PR TITLE
dev/report#16 Unreleased regression - fee levels incorrectly show sol…

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Fees are incorrectly showing as sold out, blocking change fee selection.

Before
----------------------------------------
Event selections cannot be changed as they appear sold out

After
----------------------------------------
Editability restored.

Technical Details
----------------------------------------
Fairly long tirade is over here https://lab.civicrm.org/dev/report/issues/16

Comments
----------------------------------------

### CODE REVIEW

**Review by pfigel — APPROVED** [review]
This seems like the sanest approach for now. I did a quick `r-run` where I:

- Created an event with a price set where one option had limited participants
- Confirmed that "Change Selections" was available in the participant edit form (which isn't the case on dmaster)

File: CRM/Event/Form/Participant.php [hunk 1]
```diff
@@ -258,7 +258,7 @@ public function preProcess() {
       $this->_id = NULL;
     }
     else {
-      $this->_id = CRM_Utils_Request::retrieve('id', 'Positive');
+      $this->_id = CRM_Utils_Request::retrieve('id', 'Positive', $this);
```
Comment (seamuslee001) [inline CRM/Event/Form/Participant.php]: @eileenmcnaughton @pfigel are these changes strickly needed for this it feels like it goes back to the way of not allowing opening in new tabs or something

Comment (pfigel) [reply CRM/Event/Form/Participant.php]: @seamuslee001 AIUI the other changes in this PR don't address all the issues found in https://lab.civicrm.org/dev/report/issues/16 without reverting the participant bit in #14244.

Comment (eileenmcnaughton) [reply CRM/Event/Form/Participant.php]: @seamuslee001 it's a partial revert of a change already in the rc (I didn;t revert the contribution page at this stage)

Comment (eileenmcnaughton) [reply CRM/Event/Form/Participant.php]: Specifically we are reverting this https://github.com/civicrm/civicrm-core/pull/14244/files#diff-63338fa600c17c14a171ac3033cccdedL261

**Comment by eileenmcnaughton** [comment]
Merging per @pfigel as this is a conservative approach (reverting a regression causing change). I think we addresses @seamuslee001 comment

**Pull request merged**